tests/unit/account_id_test.py

```
@@ -495,13 +632,266 @@ def test_validate_checksum_with_alias_key_set(client, alias_key):
+def test_populate_account_num_missing_account(evm_address):
+    """
+    Test that populate_account_num raises a ValueError when the mirror node
+    query does not return an account number.
+    """
+    account_id = AccountId.from_evm_address(evm_address, 0, 0)
+    mock_client = MagicMock()
+    mock_client.network.get_mirror_node_rest_url.return_value = "http://mirror_node_rest_url"
+
+    with patch("hiero_sdk_python.account.account_id.perform_query_to_mirror_node") as mock_query:
+        mock_query.return_value = {}
+        with pytest.raises(ValueError):
```

Ruff (0.14.10)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)